corgidrp/astrom.py

```diff
@@ -0,0 +1,387 @@
+import numpy as np
+
+import corgidrp.data
+
+import astropy
+import astropy.io.ascii as ascii
+from astropy.coordinates import SkyCoord
+
+import pyklip.fakes as fakes
+
+import scipy.ndimage as ndi
+import scipy.optimize as optimize
+
+def centroid(frame):
+    y, x = np.indices(frame.shape)
+    
+    ycen = np.sum(y * frame)/np.sum(frame)
+    xcen = np.sum(x * frame)/np.sum(frame)
+    
+    return xcen, ycen
+
+def shift_psf(frame, dx, dy, flux, fitsize=10, stampsize=10):
+    ystamp, xstamp = np.indices([fitsize, fitsize], dtype=float)
+    xstamp -= fitsize//2
+    ystamp -= fitsize//2
+    
+    xstamp += stampsize//2 + dx
+    ystamp += stampsize//2 + dy
+    
+    return ndi.map_coordinates(frame * flux, [ystamp, xstamp], mode='constant', cval=0.0).ravel()
+    
+def measure_offset(frame, xstar_guess, ystar_guess, xoffset_guess, yoffset_guess, guessflux=1, rad=5, stampsize=10):
+    """
+    Helper function to compute the relative offset between stars.
+    
+    Args:
```

Comment:
you'll get build errors until you fill in the docstrings each argument needs to be documented, and the return function needs to be documented.